lapack/gonum/dtrexc.go

```diff
@@ -46,31 +46,37 @@ import "gonum.org/v1/gonum/lapack"
 //
 // Dtrexc is an internal routine. It is exported for testing purposes.
 func (impl Implementation) Dtrexc(compq lapack.UpdateSchurComp, n int, t []float64, ldt int, q []float64, ldq int, ifst, ilst int, work []float64) (ifstOut, ilstOut int, ok bool) {
-	checkMatrix(n, n, t, ldt)
-	var wantq bool
-	switch compq {
-	default:
-		panic("lapack: bad value of compq")
-	case lapack.UpdateSchurNone:
-		// Nothing to do because wantq is already false.
-	case lapack.UpdateSchur:
-		wantq = true
-		checkMatrix(n, n, q, ldq)
-	}
-	if (ifst < 0 || n <= ifst) && n > 0 {
-		panic("lapack: ifst out of range")
-	}
-	if (ilst < 0 || n <= ilst) && n > 0 {
-		panic("lapack: ilst out of range")
+	switch {
+	case compq != lapack.UpdateSchur && compq != lapack.UpdateSchurNone:
+		panic(badUpdateSchurComp)
+	case n < 0:
+		panic(nLT0)
+	case ldt < max(1, n):
+		panic(badLdT)
+	case ldq < 1 || (compq == lapack.UpdateSchur && ldq < n):
```

Comment:
`case ldq < 1, compq == lapack.UpdateSchur && ldq < n:`

Reply:
Done here and everywhere else.